File: client/src/types/menu.ts
/** Contrato de GET /api/menu — espejo de Models/Menu.cs en la API. */

export interface Menu {
  restaurant: Restaurant
  categories: Category[]
}

export interface Restaurant {
  name: string
  tagline: string
  welcome: string
  description: string[]
  currency: string
}

export interface Category {
  id: string
  name: string
  items: MenuItem[]
  /** Solo bebidas: subgrupos con sus propios productos. */
  groups?: ItemGroup[]
  /** Solo pizzas: nombres de las columnas de precio. */
  sizes?: string[]
  notes?: string[]
  intro?: string[]
  note?: CategoryNote
}

export interface ItemGroup {
  name: string
  items: MenuItem[]
  note?: string
}

export interface CategoryNote {
  title: string
  options: string[]
}

export interface MenuItem {
  id: string
  name: string
  price?: number
  prices?: Record<string, number>
  description?: string
  subtitle?: string
  highlight?: boolean
}

File: src/CaballeroNegro.Api/Models/Menu.cs
namespace CaballeroNegro.Api.Models;

/// <summary>Carta completa tal como la consume el cliente React.</summary>
public sealed record Menu(Restaurant Restaurant, IReadOnlyList<Category> Categories);

public sealed record Restaurant(
    string Name,
    string Tagline,
    string Welcome,
    IReadOnlyList<string> Description,
    string Currency);

/// <summary>
/// Una sección de la carta. Según la sección, los productos van en
/// <see cref="Items"/> (la mayoría) o en <see cref="Groups"/> (bebidas).
/// </summary>
public sealed record Category(
    string Id,
    string Name,
    IReadOnlyList<MenuItem>? Items = null,
    IReadOnlyList<ItemGroup>? Groups = null,
    IReadOnlyList<string>? Sizes = null,
    IReadOnlyList<string>? Notes = null,
    IReadOnlyList<string>? Intro = null,
    CategoryNote? Note = null);

public sealed record ItemGroup(string Name, IReadOnlyList<MenuItem> Items, string? Note = null);

public sealed record CategoryNote(string Title, IReadOnlyList<string> Options);

/// <summary>
/// Un producto. Tiene <see cref="Price"/> único o <see cref="Prices"/> por tamaño (pizzas).
/// </summary>
public sealed record MenuItem(
    string Id,
    string Name,
    decimal? Price = null,
    IReadOnlyDictionary<string, decimal>? Prices = null,
    string? Description = null,
    string? Subtitle = null,
    bool Highlight = false);
